### DISCUSSION TITLE
New apscheduler release and old code portage

### DISCUSSION THREAD

**Discussion by memegauste** [agronholm/apscheduler#800] [Q&A] [answered]
Hey @agronholm !

I'm very pleased for new release and I'm waiting for the new documentation.
I'm curious because I'm still unaware how to use add_schedule / reschedule_job methods in new version.
For example I got this call in my old code and I'm still not sure how can I pass arguments in correct way:

    scheduler.add_job(
        patch_mysql_connection,
        args=[instance.submit],
        trigger='date',
        run_date=publish_date,
        id=f'newsletter_job_{instance.id}',
        replace_existing=True,
    )
My question applies also to the job rescheduling e.g.:

    scheduler.reschedule_job(
       f'newsletter_job_{instance.id}',
       trigger='date',
       run_date=publish_date,
   )
How we now pass date argument in new add_schedule method? Is there a specific trigger that does that? Also what about rescheduling?
Thanks in advance for help :)

**Comment by agronholm** [comment, marked as answer]
Try this:
```python
scheduler.add_schedule(
    patch_mysql_connection,
    DateTrigger(publish_date),
    id=f'newsletter_job_{instance.id}',
    args=[instance.submit],
    conflict_policy=ConflictPolicy.replace
)
```
